libs/index/odc/index/stac.py

```diff
@@ -207,7 +209,11 @@ def stac_transform(input_stac: Document, relative: bool = True) -> Document:
     epsg = properties['proj:epsg']
     native_crs = f"epsg:{epsg}"
 
-    geometry = _geographic_to_projected(input_stac['geometry'], native_crs)
+    # Transform geometry to the native CRS at an appropriate precision
+    pixel_size = get_in(['default', 'transform', 0], grids)
+    geometry = geom = Geometry(input_stac['geometry'], 'epsg:4326')
+    if native_crs != 'epsg:4326':
+        geometry = _geographic_to_projected(geometry, native_crs, int(1 // pixel_size))
```

Comment:
this will always set precision to `0`, I don't know what logic should be, but it's not that.

Reply:
I don't think that's true. If the pixel_size is 0.01 it'll make the precision very high.

Reply:
`//` is integer division, why are we using integer division for floating point values?

Reply:
So for `0.01` pixel size do you really need 99 points of precision?

Reply:
You're right though, it's not a good method... let me think about it a bit.